src/comparisons.jl

```diff
@@ -0,0 +1,204 @@
+- `y` -- another number (of the same numeric type than `x`)
+
+### Output
+
+A boolean that is `true` iff `x >= y`.
+
+### Algorithm
+
+This function falls back to `_leq(y, x)`. See the documentation of `_leq` for
+further details. 
+"""
+_geq(x::N, y::N) where {N<:Real} = _leq(y, x)
+
+"""
+    _geq(x::N, y::M) where {N<:Real, M<:Real}
+
+Returns `true` if the real `x` is greater or equal than `y` and false otherwise.
+
+### Input
+
+- `x` -- number
+- `y` -- another number (of possibly different numeric type than `x`)
+
+### Output
+
+A boolean that is `true` iff `x >= y`.
+
+### Algorithm
+
+This function falls back to `_leq(y, x)`, with type promotion if needed. See the
+documentation of `_leq` for further details.
+"""
+_geq(x::N, y::M) where {N<:Real, M<:Real} = _geq(promote(x, y)...)
+
+"""
+    isapproxzero(x::N; kwargs...) where {N<:Real}
+
+Returns `true` if the real `x` is greater or equal than `y` and false otherwise.
+
+### Input
+
+- `x`      -- number
+- `kwargs` -- ignored
+ 
+### Output
+
+A boolean that is `true` iff `x ≈ 0`.
+
+### Algorithm
+
+This is a fallback implementation for any real `x` such that `x ≈ 0` is `true`
+whenever `x` is equal to zero in the same numeric type as `x`.
+"""
+function isapproxzero(x::N; kwargs...) where {N<:Real}
+    return x == zero(N)
+end
+
+"""
+    isapproxzero(x::N; ztol::Real=ABSZTOL(N)) where {N<:AbstractFloat}
+
+Returns `true` if `x` is approximately zero `false` otherwise.
+
+### Input
+
+- `x`    -- number
+- `ztol` -- (optional, default: `ABSZTOL`) tolerance against zero
+
+### Output
+
+A boolean that is `true` iff `x ≈ 0`.
+
+### Algorithm
+
+It is considered that `x ≈ 0` whenever `x` (in absolute value) is smaller than the
+tolerance for zero, `ztol`.
+"""
+function isapproxzero(x::N; ztol::Real=ABSZTOL(N)) where {N<:AbstractFloat}
+    return abs(x) < ztol
+end
+
+"""
+    _isapprox(x::N, y::N;
+              rtol::Real=Base.rtoldefault(N),
+              ztol::Real=ABSZTOL(N),
+              atol::Real=zero(N)) where {N<:AbstractFloat}
+
+Returns `true` if `x` is approximately equal than `y` and false otherwise.
+
+### Input
+
+- `x`    -- number
+- `y`    -- another number (of the same numeric type than `x`)
+- `rtol` -- (optional, default: `Base.rtoldefault(N)`) relative tolerance
+- `ztol` -- (optional, default: `ABSZTOL(N)`) absolute tolerance for comparison against zero
+- `atol` -- absolute tolerance
+
+### Output
+
+A boolean that is `true` iff `x ≈ y`.
+
+### Algorithm
+
+This comparison is split into both `x` and `y` approximately zero checked using
+`isapproxzero(x, y)`, or `x ≈ y` checked using Julia's `isapprox(x, y)`. In the
+latter function we use zero absolute tolerance and `rtol` relative tolerance.
+"""
+function _isapprox(x::N, y::N;
+                   rtol::Real=Base.rtoldefault(N),
+                   ztol::Real=ABSZTOL(N),
+                   atol::Real=zero(N)) where {N<:AbstractFloat}
+    if isapproxzero(x, ztol=ztol) && isapproxzero(y, ztol=ztol)
```

Comment:
I do not understand the motivation for this function.
Why should both numbers being close to zero be a special case? If both are close to ten, I would expect the same result.
I had expected something like `if isapproxzero(x-y, ztol=ztol)`.

Reply:
Comparing to zero with default tolerances is a special case in Julia's `isapprox`, see the last paragraph in `?isapprox`. This function tries to combine `isapprox` with its default values and a branch for `x ≈ y ≈ 0` which includes `x == y == 0` but also admits a tolerance `ztol`.

Reply:
> the argument atol is not used,

`atol` is passed to `isapprox`. We use an absolutel-like comparison (with `atol=TOL`) for example [here](https://github.com/JuliaReach/LazySets.jl/blob/master/src/Approximations/iterative_refinement.jl#L89). (we can make this check more robust by using a relative comparison or a combination or relative and absolute).

Reply:
😮  below it should read

```julia
return isapprox(x, y, rtol=rtol, atol=atol)
```